packages/keyring-eth-ledger-bridge/src/ledger-error-handler.test.ts

```diff
@@ -0,0 +1,200 @@
+      expect((thrownError as LedgerStatusError).statusCode).toBe(0x6a80);
+      expect((thrownError as LedgerStatusError).message).toBe(
+        'Ledger: Blind signing must be enabled',
+      );
+    });
+
+    it('should handle status code 0x5515 (device locked)', () => {
+      const error = createTransportStatusError('Device locked', 0x5515);
+
+      expect(() => handleLedgerTransportError(error, fallbackMessage)).toThrow(
+        LedgerStatusError,
+      );
+
+      let thrownError: unknown;
+      try {
+        handleLedgerTransportError(error, fallbackMessage);
+      } catch (error_: unknown) {
+        thrownError = error_;
+      }
+      expect(thrownError).toBeInstanceOf(LedgerStatusError);
+      expect((thrownError as LedgerStatusError).statusCode).toBe(0x5515);
+      expect((thrownError as LedgerStatusError).message).toBe(
+        'Ledger: Device is locked. Unlock it to continue.',
+      );
+    });
+
+    it('should handle status code 0x650f (app closed)', () => {
+      const error = createTransportStatusError('App closed', 0x650f);
+
+      expect(() => handleLedgerTransportError(error, fallbackMessage)).toThrow(
+        LedgerStatusError,
+      );
+
+      let thrownError: unknown;
+      try {
+        handleLedgerTransportError(error, fallbackMessage);
+      } catch (error_: unknown) {
+        thrownError = error_;
+      }
+      expect(thrownError).toBeInstanceOf(LedgerStatusError);
+      expect((thrownError as LedgerStatusError).statusCode).toBe(0x650f);
+      expect((thrownError as LedgerStatusError).message).toBe(
+        'Ledger: Ethereum app closed. Open it to unlock.',
+      );
+    });
+
+    it('should handle unknown status codes by preserving original message', () => {
+      const unknownStatusCode = 0x9999;
+      const originalMessage = 'Unknown transport error';
+      const error = createTransportStatusError(
+        originalMessage,
+        unknownStatusCode,
+      );
+
+      expect(() => handleLedgerTransportError(error, fallbackMessage)).toThrow(
+        LedgerStatusError,
+      );
+
+      let thrownError: unknown;
+      try {
+        handleLedgerTransportError(error, fallbackMessage);
+      } catch (error_: unknown) {
+        thrownError = error_;
+      }
+      expect(thrownError).toBeInstanceOf(LedgerStatusError);
+      expect((thrownError as LedgerStatusError).statusCode).toBe(
+        unknownStatusCode,
+      );
+      expect((thrownError as LedgerStatusError).message).toBe(originalMessage);
+    });
+  });
+
+  describe('when error is not TransportStatusError', () => {
+    it('should re-throw Error instances as-is', () => {
+      const originalError = new Error('Original error message');
+
+      expect(() =>
+        handleLedgerTransportError(originalError, fallbackMessage),
+      ).toThrow(originalError);
+
+      expect(() =>
+        handleLedgerTransportError(originalError, fallbackMessage),
+      ).toThrow('Original error message');
```

Comment:
I guess we could just re-use the underlying message here.

```suggestion
      ).toThrow(error.message);
```